include/fft_helpers.h

```diff
@@ -28,44 +29,75 @@
 
 #include "lmms_export.h"
 
+#include <vector>
 #include <fftw3.h>
 
+// NOTE: FFT_BUFFER_SIZE should be considered deprecated!
+// It is used by Eq plugin and some older code here, but this should be a user
+// switchable parameter, not a constant. Use a value from FFT_BLOCK_SIZES
 const int FFT_BUFFER_SIZE = 2048;
 
-enum WINDOWS
-{
-        KAISER=1,
-        RECTANGLE,
-        HANNING,
-        HAMMING
+// Allowed FFT block sizes. Ranging from barely useful to barely acceptable
+// because of performance reasons.
+const std::vector<int> FFT_BLOCK_SIZES = {256, 512, 1024, 2048, 4096, 8192, 16384};
+
+// List of FFT window functions supported by precomputeWindow()
+enum FFT_WINDOWS {
+		RECTANGULAR = 0,
+		BLACKMAN_HARRIS,
+		HAMMING,
+		HANNING
 };
 
-/* returns biggest value from abs_spectrum[spec_size] array
+
+/* Returns biggest value from abs_spectrum[spec_size] array.
```

Comment:
I just barely know Doxygen exists, I never really used it. :) It feels to me it only adds "noise" to the comments, making them harder to read.
In this case, I just followed the previously established comment style of the file.
Otherwise I usually explain what the function does in a sentence or two, allowing me to easily add extra notes if something needs extra attention, and to save space if something is entirely obvious.

Reply:
I never use doxygen's created HTMLs either, because you can be much faster by just navigating to the function in question using your IDE. However, some IDEs (like QtCreator) can parse doxygen. E.g. if I hold my mouse over a function, I see the definition and and the doxygen description in a tooltip. Your normal comments would not show up in QtCreator.

Reply:
Right. I keep forgetting about the IDE argument. I just use gvim to write the code, one terminal to compile and one to manage git, so it never occurs to me Doxygen can be actually useful for something. :)
I added the `@return` tags in fft_helpers, since it was already more or less using it. These functions are also supposed to be shared by multiple plugins, so the extra documentation features probably make sense here.

Reply:
Sorry, it's not yet a Doxygen comment. Doxygen comments need to be introduced like
```c++
/**
```
, i.e. with 2 stars. That makes doxygen/QtCreator understand that it's a comment for the following declaration. (or for inline, use `//!`). Can you please fix that?

Reply:
Oops, sorry. ^^; Will do.

Reply:
The latest fix of the previous two fixes should be in place... :D